portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -19,14 +19,73 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import com.google.gson.Gson;
+import java.util.ArrayList;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import java.util.Collections;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-
+  ArrayList <String> comments = new ArrayList <String> ();
+  ArrayList <String> comments_fin = new ArrayList <String> ();
+    
+    
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    
     response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    int i = 0;
+    int l = comments_fin.size();
+    while(i<l){
+      if(comments_fin.get(i)==null){
+        comments_fin.remove(i);
+        l--;
+      }
+      else{
+        i++;
+      }
+
+    }
+    
+    response.getWriter().println(comments_fin);
+  }
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String comment = request.getParameter("comment");
+    comments_fin.clear();
```

Comment:
If I understand correctly, we're storing state in the servlet's fields. Can we fix that? What state is needed to persist between requests? Ideally each request can stand on its own, and doesn't impact any other requests.